src/source/Signaling/LwsApiCalls.c

```diff
@@ -2171,14 +2367,28 @@ PVOID receiveLwsMessageWrapper(PVOID args)
     // Updating the diagnostics info before calling the client callback
     ATOMIC_INCREMENT(&pSignalingClient->diagnostics.numberOfMessagesReceived);
 
+    if (messageType == SIGNALING_MESSAGE_TYPE_OFFER) {
+        MUTEX_LOCK(pSignalingClient->offerSendReceiveTimeLock);
+        pSignalingClient->offerReceivedTime = GETTIME();
+        MUTEX_UNLOCK(pSignalingClient->offerSendReceiveTimeLock);
+
+        if (pSignalingClient->mediaStorageConfig.storageStatus) {
+            MUTEX_LOCK(pSignalingClient->jssWaitLock);
+            ATOMIC_STORE_BOOL(&pSignalingClient->offerReceived, TRUE);
+            DLOGI("Offer Received from JoinStorageSession Call.");
+            pSignalingClient->diagnostics.joinSessionToOfferRecvTime =
```

Comment:
Should `PROFILE_WITH_START_TIME_OBJ` be used as is below for the SIGNALING_MESSAGE_TYPE_ANSWER case?

Reply:
That macro itself calls GETTIME().  The issue is we need to acquire the jssWaitLock before we can set that value and I was afraid if there's lock contention then the actual time the offer was received may not get stored and we'll have an inaccurate value there, hence why I set the offer received time *before* acquiring the lock.